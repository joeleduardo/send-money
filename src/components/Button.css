button {
  outline: transparent;
  background: #4b3cfa;
  color: #fff;
  margin-top: 6rem;
  padding: 1rem;
  border-radius: 8px;
  width: 100%;
  font-size: 1rem;
  cursor: pointer;
}