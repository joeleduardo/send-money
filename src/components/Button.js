import React from 'react';
import PropTypes from 'prop-types';
import './Button.css'

const Button = (props) => {
  const {
    text,
    enabled,
    onClick,
  } = props;

  const style = !enabled ? { backgroundColor: '#efefef', cursor: 'auto', color: '#ccc' } : {};

  return (
    <button onClick={e => onClick(e)} style={style} disabled={!enabled}>
      {text}
    </button>
  );
};

Button.propTypes = {
  text: PropTypes.string.isRequired,
  enabled: PropTypes.bool,
  onClick: PropTypes.func
};

Button.defaultProps = {
  text: '',
  enabled: true
};

export default Button;
