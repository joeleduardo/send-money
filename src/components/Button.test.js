import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

test('Button component enabled', () => {
  const mockClick = jest.fn();
  const cButton = shallow(<Button test="Send money" enabled onClick={(e) => mockClick(e)}/>);
  cButton.simulate('click');
  expect(mockClick).toHaveBeenCalled();
  expect(cButton).toMatchSnapshot();
});

test('Button component enabled', () => {
  const mockClick = jest.fn();
  const cButton = shallow(<Button test="Send money" enabled={false} onClick={(e) => mockClick(e)}/>);
  expect(mockClick.mock.calls.length).toBe(0);
  expect(cButton).toMatchSnapshot();
});

