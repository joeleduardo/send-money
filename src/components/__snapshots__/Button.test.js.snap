// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component enabled 1`] = `
<button
  disabled={false}
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`Button component enabled 2`] = `
<button
  disabled={true}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#efefef",
      "color": "#ccc",
      "cursor": "auto",
    }
  }
/>
`;
